spectator-ext-spark/src/main/resources/reference.conf

```diff
@@ -1,26 +1,38 @@
 
 spectator.spark {
   name-patterns = [
-    // EXECUTORS
+
+    // EXECUTORS Spark 1.6
     // 97278898-4bd4-49c2-9889-aa5f969a7816-0013.97278898-4bd4-49c2-9889-aa5f969a7816-S1/2.executor.filesystem.file.largeRead_ops
     {
-      pattern = "^([^.]+)\\.(([^.]+)\/)?(\\d+)\\.((executor|jvm|CodeGenerator)\\..*)"
-      name = 5
+      pattern = "^([^.]+)\\.([^.]+)\\/(\\d+)\\.((executor|jvm|CodeGenerator)\\..*)"
+      name = 4
       tags = {
         "appId" = 1
-        "agentId" = 3
-        "executorId" = 4
+        "agentId" = 2
+        "executorId" = 3
       }
     },
 
+    // EXECUTORS Spark 2.0/2.1
+    // 97278898-4bd4-49c2-9889-aa5f969a7816-0013.2.executor.filesystem.file.largeRead_ops
+    {
+      pattern = "^([^.]+)\\.(\\d+)\\.((executor|jvm|HiveExternalCatalog|CodeGenerator)\\..*)"
```

Comment:
On second look, the regex seems inconsistent with the example (here and the original). From the sample values it looks as though `executor` is first and required, followed by one of the following values. However, the regex says one of `executor`, `jvm`, `HiveExternalCatalog `, or `CodeGenerator`, followed by `.` and then anything.

Seems we may be missing some negative test cases.

Please let me know if there's something I'm missing.  If not, please correct and add negative tests for the 2.+ cases.

If you want to fix and add negative tests for 1.6, happy to accept the contribution :-), but not requesting that of you (one of us on telemetry can fill those in).

Reply:
I do not understand, do you want me to enumerate all possible examples for metrics emitted by the executors?

Reply:
Sorry it wasn't clear. The additional example IDs you provided help.

I'm not familiar with the format of these IDs and my google search for a format spec didn't turn anything up (is there one available somewhere?). From the original example ids, it seemed like the pattern should have been specified as:
,,,
      pattern = "^([^.]+)\\.(\\d+)\\.(executor)\\.((jvm|HiveExternalCatalog|CodeGenerator)\\..*)"
,,,

However, the additional example IDs you provided clear that up and show that it is correct as is. So, thanks for adding those.